addon/adapters/osf-adapter.js

```diff
@@ -10,14 +11,54 @@ export default DS.JSONAPIAdapter.extend(DataAdapterMixin, {
     authorizer: 'authorizer:osf-token',
     host: config.OSF.apiUrl,
     namespace: config.OSF.apiNamespace,
-
-    buildURL(modelName, id, snapshot, requestType, query) {  // jshint ignore:line
-        var url = this._super(...arguments);
+    buildURL(modelName, id, snapshot, requestType, query, dirtyRelationship) { // jshint ignore:line
+        var url;
+        var links = dirtyRelationship ? snapshot.record.get(
+            `links.relationships.${Ember.String.underscore(dirtyRelationship)}.links`
+        ) : false;
+        if (links) {
+            url = links.self ? links.self.href : links.related.href;
+        } else {
+            url = this._super(...arguments);
+        }
         // Fix issue where CORS request failed on 301s: Ember does not seem to append trailing
         //  slash to URLs for single documents, but DRF redirects to force a trailing slash
-        if (url.lastIndexOf('/') !== 0) {
+        if (url.lastIndexOf('/') !== url.length - 1) {
```

Comment:
Er, that'd also be weird. See this page for a suggested endsWith polyfill:
http://stackoverflow.com/questions/280634/endswith-in-javascript

...I originally tried using ES6 `endsWith`, but encountered errors because it was supported in chrome, but not phantomJS. As long as the intent is preserved, I'm happy with whatever works.